### ISSUE TITLE
Detect if a stacktraceframe is library_frame or not.

### ISSUE THREAD

**Issue by simitt** [elastic/apm-server#426]
Set boolean `library_frame` for every stacktrace frame if not set.

Allow for configurable regexp indicating a frame is a `library_frame`. 

@jahtalab will look into whether default values make sense if no config is given.

**Comment by beniwohli** [comment]
This should only be done for data from frontend agents, right?

**Comment by graphaelli** [comment]
@beniwohli That's right, this is only while applying sourcemaps within apm-server

**Comment by simitt** [comment]
@jahtalab could you please comment your findings here.

**Comment by jahtalab** [comment]
The spec doesn't say much about this, [Source Map Revision 3 Proposal - Google Docs](https://docs.google.com/document/d/1U1RGAehQwRypUTovF1KRlpiOFze0b-_2gc6fAH0KY0k/edit#)

But looking into webpack implementation, they have this template:
`devtoolModuleFilenameTemplate: "webpack:///[resource-path]?[loaders]"`

The default template should generate path with "node_modules" part, But webpack used to replace "node_modules" with "~/"

Related issues:
[Document a proper way to debug applications bundled by webpack · Issue #110 · webpack/webpack.js.org](https://github.com/webpack/webpack.js.org/issues/110)

[Document `webpack://` & make sourcemaps have relative paths to original files · Issue #3603 · webpack/webpack](https://github.com/webpack/webpack/issues/3603)

[How to resolve file path from a webpack:/// URI · Issue #2395 · webpack/webpack](https://github.com/webpack/webpack/issues/2395)

In conclusion there are different templates for how the final path of the file is generated and in the end developers can change it but currently the same pattern we had in [opbeat](https://github.com/opbeat/opbeat/blob/2c3fe34712027d9afc1594fb12c5d68ee75f8e45/webapp/app/stacks.py#L457) is still valid.

I suggest to use opbeat's library pattern as default but make it configurable.
